src/main/webapp/assets/modules/get-promoted/userPromotionJobInfoDirective.js

```diff
@@ -0,0 +1,38 @@
+techlooper.directive("userPromotionJobInfo", function ($http, userPromotionService, $location, utils, $translate) {
+  return {
+    restrict: "E",
+    replace: true,
+    templateUrl: "modules/get-promoted/userPromotionJobInfo.html",
+    link: function (scope, element, attr, ctrl) {
+      scope.$on("stateChanged", function () {
+        delete scope.promotionEmail;
+        scope.promotionEmailForm.$setPristine();
+        scope.promotionEmailForm.$setUntouched();
+        utils.removeRedundantAttrs(scope.promotionEmailForm, ["$sentEmail"]);
+      });
+
+      scope.sendPromotionEmail = function () {
+        if (!scope.promotionEmailForm.$valid) {
+          return false;
+        }
+
+        var promotionInfo = angular.copy(scope.masterPromotion);
+        if (promotionInfo.jobLevelIds.length === 0) delete promotionInfo.jobLevelIds;
+
+        var promotionResult = scope.masterPromotion.result;
+        utils.removeRedundantAttrs(promotionInfo, ["result", "jobCategoryTitle", "jobLevelTitle", "jobLevelId"]);
+        $http
+          .post("getPromoted/email", {
+            getPromotedRequest: promotionInfo,
+            lang: $translate.use(),
+            email: scope.promotionEmail,
+            hasResult: promotionResult && promotionResult.totalJob > 0
+          })
+          .success(function (data, status, headers, config) {
```

Comment:
'config' is defined but never used.<br>'headers' is defined but never used.<br>'status' is defined but never used.<br>'data' is defined but never used.